charts/base.go

```diff
@@ -47,10 +46,72 @@ type BaseConfiguration struct {
 	// ParallelAxisList represents the component list which is the coordinate axis for parallel coordinate.
 	ParallelAxisList []opts.ParallelAxis
 
-	HasGeo        bool `json:"-"`
-	HasRadar      bool `json:"-"`
-	HasParallel   bool `json:"-"`
-	HasSingleAxis bool `json:"-"`
+	has3DAxis     bool
+	hasXYAxis     bool
+	hasGeo        bool
+	hasRadar      bool
+	hasParallel   bool
+	hasSingleAxis bool
+}
+
+func (bc *BaseConfiguration) JSON() map[string]interface{} {
+	obj := map[string]interface{}{
+		"title":   bc.Title,
+		"legend":  bc.Legend,
+		"tooltip": bc.Tooltip,
+		"series":  bc.MultiSeries,
+	}
+
+	if bc.hasGeo {
+		obj["geo"] = bc.GeoComponent
+	}
+
+	if bc.hasRadar {
+		obj["radar"] = bc.RadarComponent
+	}
+
+	if bc.hasParallel {
+		obj["parallel"] = bc.ParallelComponent
+		obj["parallelAxis"] = bc.ParallelAxisList
+	}
+
+	if bc.hasSingleAxis {
+		obj["singleAxis"] = bc.SingleAxis
+	}
+
+	if bc.Toolbox.Show {
+		obj["toolbox"] = bc.Toolbox
+	}
+
+	if len(bc.DataZoomList) > 0 {
+		obj["dataZoom"] = bc.DataZoomList
+	}
+
+	if len(bc.VisualMapList) > 0 {
+		obj["visualMap"] = bc.VisualMapList
+	}
+
+	if bc.hasXYAxis {
+		obj["xAxis"] = bc.XAxisList
+		obj["yAxis"] = bc.YAxisList
+	}
+
+	if bc.has3DAxis {
+		obj["xAxis3D"] = bc.XAxis3D
+		obj["yAxis3D"] = bc.YAxis3D
+		obj["zAxis3D"] = bc.ZAxis3D
+		obj["grid3D"] = bc.Grid3D
+	}
+
+	if bc.Theme == "white" {
+		obj["color"] = bc.Colors
+	}
+
```

Comment:
should be `bc.Theme != 'white'` ？

Reply:
Only valid when the theme is white because others theme use their style

```diff
--    {{- if eq .Theme "white" }}	
--        color: {{ .Colors }},	
--    {{- end }}
```

Reply:
gotcha.